scripts/migration/migrate-addon.sh

```diff
@@ -0,0 +1,653 @@
+      "volumeMounts": [{"mountPath": "/var/run/root-certs", "name": "meshca-root", "readOnly": true}]
+    }],
+    "volumes": [{"name": "meshca-root", "secret":{"secretName": "meshca-root"}}]
+  }}}}'
+  echo -e "${green}OK${clr}"
+
+  # Wait for Citadel to distribute the cert everywhere. Note: this always takes at least one minute
+  # due to a bug/feature in Citadel causing it to delay provisioning at startup.
+  heading "Waiting for root certificate to distribute to all pods. This will take a few minutes..."
+  for namespace in $(kubectl get namespaces -oname | cut -d/ -f2); do
+    # Check against a portion of the root to see if its been distributed.
+    # Note: this doesn't ensure the workloads have actually hot-restarted to pickup the change.
+    ROOT_CERT="$(kubectl get secret -n "${namespace}" istio.default -ojsonpath='{.data.root-cert\.pem}')"
+    LAST_ERR=$?
+    if [[ "${LAST_ERR}" != 0 ]]; then
+      # Likely namespace was deleted while in progress
+      continue
+    fi
+    while ! echo "${ROOT_CERT}" | base64 -d | grep -q 8zyiCcMm1i2yVVphwE1kczFwunAh0JB896VaXGVxXeKEAMQoXHjgDdCYp8; do
+      echo "ASM root certificate not distributed to ${namespace}, trying again later"
+      sleep 3
+      ROOT_CERT="$(kubectl get secret -n "${namespace}" istio.default -ojsonpath='{.data.root-cert\.pem}')"
+      LAST_ERR=$?
+      if [[ "${LAST_ERR}" != 0 ]]; then
+        # Likely namespace was deleted while in progress
+        continue
+      fi
+    done
+    echo -e "${green}ASM root certificate distributed to namespace ${namespace}${clr}"
+  done
+  # Wait 75 seconds. Kubelet runs a sync loop on a 1 minute interval, so its unlikely to take more than 75s
+  # ideally we have a better wait here, but its hard to do. We can exec in and read the file or hit /certs,
+  # but both would be slow and require a lot of code to write.
+  if [[ "${SKIP_WAIT_FOR_ROOT:-}" == "" ]]; then
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+    echo "Waiting for proxies to pick up the new root certificate..."; sleep 15
+  fi
+  echo -e "${green}OK${clr}"
+}
+
+disable_galley_webhook() {
+  prompt "Replacing the Istio validation webhook with an Anthos Service Mesh validation webhook..."
+  kube patch clusterrole -n istio-system istio-galley-istio-system --type='json' -p='[{"op": "replace", "path": "/rules/2/verbs/0", "value": "get"}]'
+  kube delete ValidatingWebhookConfiguration istio-galley --ignore-not-found
```

Comment:
Is this possible to roll back?

Reply:
we now store the configuration before deleting it..the rollback can rely on that if needed